packages/mongodb-chatbot-ui/src/services/conversations.ts

```diff
@@ -62,12 +62,25 @@ export class TimeoutError<Data extends object = object> extends Error {
   }
 }
 
+/**
+  Options to include with every fetch request made by the ConversationService.
+  This can be used to set headers, etc.
+ */
+export type ConversationFetchOptions = Omit<
+  RequestInit,
+  "body" | "method" | "headers" | "signal"
+> & {
+  headers?: Headers;
+};
+
 export type ConversationServiceConfig = {
   serverUrl: string;
+  fetchOptions?: ConversationFetchOptions;
 };
 
 export class ConversationService {
   private serverUrl: string;
+  private fetchOptions?: ConversationFetchOptions;
```

Comment:
We define `this.fetchOptions` in the class constructor so it should always be defined right?

Maybe an opportunity for cleanup in other spots in the class where we currently assume `fetchOptions` may be undefined and use nullish coalescing (i.e. `??`)